### PR TITLE
Move ANSI HTML-ifying to the site

### PR DESCRIPTION
**Pull request by Xyene**
Addresses #478.

Should not be merged until the appropriate site-side change is made.

### CODE REVIEW

File: dmoj/utils/ansi.py [hunk 1]
```diff
@@ -29,3 +26,7 @@ def format_inline(text, attrs):
 
     return re.sub(r'#ansi\[(.*?)\]\((.*?)\)',
                   lambda x: format_inline(x.group(1), x.group(2)) if not no_ansi else x.group(1), text)
+
+
+def print_ansi(*args, **kwargs):
+    print(*list(map(ansi_style, args)), **kwargs)
```
Comment (quantum5) [inline dmoj/utils/ansi.py]: Do you need the `list`?

```pycon
$ python3
Python 3.7.3 (default, Apr  3 2019, 05:39:12) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> print(*map(int, '123'))
1 2 3
```

**Comment by Xyene** [comment]
> 
> 
> I don't see how `print_ansi` is actually relevant to this change. Perhaps it would be better to make it a separate PR...

Perhaps, but it was bugging my autoformatter and this'll be rebased anyway.

File: dmoj/utils/ansi.py [hunk 2]
```diff
@@ -29,3 +26,7 @@ def format_inline(text, attrs):
 
     return re.sub(r'#ansi\[(.*?)\]\((.*?)\)',
                   lambda x: format_inline(x.group(1), x.group(2)) if not no_ansi else x.group(1), text)
+
+
+def print_ansi(*args, **kwargs):
+    print(*map(ansi_style, args), **kwargs)
```
Comment (quantum5) [inline dmoj/utils/ansi.py]: Wait, what happens if there are non-string arguments?

File: dmoj/judge.py [hunk 3]
```diff
@@ -183,7 +183,8 @@ def begin_grading(self, id, *args, **kwargs):
         self.current_submission_thread.daemon = True
         self.current_submission_thread.start()
 
-        if is_blocking:
+        # Submission may have been killed already, but block if it hasn't been.
```
Comment (quantum5) [inline dmoj/judge.py]: What does this have to do with ANSI escape codes?

Comment (Xyene) [reply dmoj/judge.py]: It's a race. Not HTMLifying makes it trigger all the time and crashes the judge, so it is fixed here.

File: dmoj/utils/ansi.py (hunk 2 above)
Comment (Xyene) [reply dmoj/utils/ansi.py]: There are currently none, and I don't want to introduce more unused code. It's trivially fixable if we ever need to print non-string things.
